File: frontend/medops/src/types/events.ts
// src/types/events.ts
// =====================================================
// SEVERIDAD
// =====================================================
export type EventSeverity = "info" | "warning" | "critical";
// =====================================================
// EVENTO DE AUDITORÍA
// =====================================================
export interface Event {
  id: number;
  
  // Institución y usuario
  institution?: number | null;
  actor_user?: number | null;
  actor_name?: string | null;
  
  // Entidad y acción
  entity: string;
  entity_id: number;
  action: string;
  
  // Metadata
  metadata?: Record<string, any>;
  severity: EventSeverity;
  notify: boolean;
  is_read?: boolean;
  
  // Campos enriquecidos del backend
  title?: string;
  description?: string;
  category?: string;
  action_label?: string;
  action_href?: string | null;
  badge_action?: "create" | "update" | "delete" | "other";
  
  // Timestamp
  timestamp: string;
  
  // Display
  created_at?: string;
  updated_at?: string;
}
// =====================================================
// CREATE EVENT INPUT
// =====================================================
export interface CreateEventInput {
  title: string;
  description: string;
  date: string;
}

File: core/utils/events.py
from core.models import Event
def log_event(
    entity: str, 
    entity_id: int, 
    action: str, 
    actor: str = "", 
    metadata: dict | None = None,
    notify: bool = False
):
    """
    Registra un evento de auditoría en la base de datos.
    
    Args:
        entity (str): Nombre de la entidad (ej. "Patient", "Appointment", "Payment").
        entity_id (int): ID del objeto relacionado.
        action (str): Acción realizada (ej. "create", "update", "delete").
        actor (str): Usuario o sistema que ejecuta la acción.
        metadata (dict): Información adicional opcional.
        notify (bool): Si True, el evento se muestra como notificación.
    """
    Event.objects.create(
        entity=entity,
        entity_id=entity_id,
        action=action,
        actor_name=actor,
        metadata=metadata or {},
        notify=notify
    )